Seminars/Sem.04/Pract.04/Task04.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#pragma warning (disable : 4996)
+const char MESSAGE_UKNOWN[] = "Uknown";
+
+class Secret {
+private:
+	char* task;
+	char* password;
+	int failedLogins;
+
+	bool DoPasswordsMatch(const char* password) const;
+
+public:
+	Secret();
+	Secret(const char* task, const char* password);
+
+	const char* GetPassword() const;
+	int GetFailedLogins() const;
+
+	void SetTask(const char* task);
+	void SetPassword(const char* password);
+
+	const char* GetTask(const char* password);
+	void SetPassword(const char* newPassword, const char* oldPassword);
+	void SetTask(const char* newTask, const char* password);
+
+	~Secret();
+};
+
+Secret::Secret() : Secret(MESSAGE_UKNOWN, MESSAGE_UKNOWN) {}
+
+Secret::Secret(const char* task, const char* password) {
+	this->task = new char[strlen(task) + 1];
+	this->password = new char[strlen(password) + 1];
+	SetTask(task);
+	SetPassword(password);
+}
+
+const char* Secret::GetPassword() const {
+	return password;
+}
+
+int Secret::GetFailedLogins() const {
+	return failedLogins;
+}
+
+void Secret::SetTask(const char* task) {
+	strcpy(this->task, task);
+}
+
+void Secret::SetPassword(const char* password) {
```

Comment:
We need checks in setters.
- if the parameter task is nullptr, what will happen?
- Should we delete our data before that (this->task)?
- How do we know if we are successful ? (boolean)